### PR TITLE
Product listing

### PR DESCRIPTION
**Pull request by raghav2110**
Hi , I have added filter and product lisiting page in my ecom. Please review it

### CODE REVIEW

**Review by Hamdrive — COMMENTED** [review]
Good work done, code looks clean and proper logic implemented.
Will suggest to properly format your code to improve readability!

File: src/context/product-context.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const ProductProvider = ({ children }) => {
+    const [products, setProducts] = useState([]);
+    const [state, dispatch] = useReducer(applyFilters, filterSettings);
+    const [categories, setCategories] = useState([]);
+
+    useEffect(() => {
+        (async () => {
+          try {
+            const res = await axios.get("/api/categories");
+            console.log(res.data.categories)
```
Comment (adith-a-danthi) [inline src/context/product-context.jsx]: You can remove console log statements

File: src/pages/Products/ProductList.jsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                            <div className="footer-card">
+                                <div className="name-comp-card">
+                                    <button className="btn-addToCart">Add To Cart</button>
+                                </div>
+                                <div className="icons-comp">
+                                    <i className="fa fa-heart icon"></i>
+                                </div>
+                            </div>
+                        </div>
+                    </a>)}
```
Comment (adith-a-danthi) [inline src/pages/Products/ProductList.jsx:36]: Product card can be moved into a  separate component.

**Review by adith-a-danthi — COMMENTED** [review]
Clean code and logic.  Some formatting would help would help with readability.
**Suggestion** : `filterModule.js` and `productFilterList.js` could be clubbed into a single file, sam for `applyFilter` and `filterSetting`

File: src/context/product-context.jsx (hunk 1 above)
Comment (raghav2110) [reply src/context/product-context.jsx]: Thanks for bringing that to notice will remove that.

**Pull request merged**